python/s3_cross_account_access.py

```diff
@@ -0,0 +1,377 @@
+# not use this file except in compliance with the License. A copy of the License is located at
+#
+#        http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+
+import json
+import datetime
+import boto3
+import botocore
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::::Account'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+#############
+# Main Code #
+#############
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    """Form the evaluation(s) to be return to Config Rules
+
+    Return either:
+    None -- when no result needs to be displayed
+    a string -- either COMPLIANT, NON_COMPLIANT or NOT_APPLICABLE
+    a dictionary -- the evaluation dictionary, usually built by build_evaluation_from_config_item()
+    a list of dictionary -- a list of evaluation dictionary , usually built by build_evaluation()
+
+    Keyword arguments:
+    event -- the event variable given in the lambda handler
+    configuration_item -- the configurationItem dictionary in the invokingEvent
+    valid_rule_parameters -- the output of the evaluate_parameters() representing validated parameters of the Config Rule
+
+    Advanced Notes:
+    1 -- if a resource is deleted and generate a configuration change with ResourceDeleted status, the Boilerplate code will put a NOT_APPLICABLE on this resource automatically.
+    2 -- if a None or a list of dictionary is returned, the old evaluation(s) which are not returned in the new evaluation list are returned as NOT_APPLICABLE by the Boilerplate code
+    3 -- if None or an empty string, list or dict is returned, the Boilerplate code will put a "shadow" evaluation to feedback that the evaluation took place properly
+    """
+################################################################################################################################
+    ###############################
+    # Add your custom logic here. #
+    ###############################
+
+	# set the flga with APPLICABLE
+	flag = 'APPLICABLE'
+	# Initialice clinet to access s3
+	s3client = boto3.client('s3')
+	# get the list of buckets of the account.
+	BucketsList = s3client.list_buckets()
+	# Check whthere each bucket has privided to access from different AWS account.
+	# 1. get a bucket from the list.
+	# 2. get the acl form the bucket.
+	# 3. check whether owner ID is equal to grantee Id.
+	#   If it equal, the bucket is applicable.
+	#   If not, the bucket is not applicable.
+	for buck in BucketsList['Buckets']:

```

Comment:
Use the configuration_item to get the bucket name, no need for the list. This rule is probably configuration change triggered.